File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        // [1, 2, 4, 6]
        let res = Array(nums.length).fill(0);
        for(const n in nums) {
            let prod = ((n === 0) ? 0 : 1);
            if(prod != 0) {
                for(const m in nums) {
                    if(nums[n] === nums[m]) {
                        continue;
                    }
                    prod = prod * nums[m];
                }
            }
            res[n] = prod;
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        l = 0
        res = 0
        if len(s) == 1:
            return 1
        for r in range(1, len(s) - 1):
            if s[r] not in seen:
                res += 1
                seen.add(s[r])
            else:
                l = r + 1
        return res
        